core/state/snapshot/difflayer.go

```diff
@@ -73,23 +74,14 @@ var (
 	// bloom key for an account/slot. This is randomized at init(), so that the
 	// global population of nodes do not all display the exact same behaviour with
 	// regards to bloom content
-	bloomDestructHasherOffset = 0
-	bloomAccountHasherOffset  = 0
-	bloomStorageHasherOffset  = 0
+	bloomAccountHasherOffset = 0
```

Comment:
A question for @holiman , do you know why we have to randomize these offsets?

I can understand we must choose different offsets for account and destructs,
but what if we remove the destructs, is it still necessary to select a different offset
for account data and storage data?

Reply:
The bloom is an optimization. If there ever is a block or situation where the bloom notoriously underperforms, either due to random or because of some sort of attack, it is better if only a small part of all nodes are affected, globally. 

In places where we can, I think we should use such tricks in order to make the macro effects of bugs less dangerous. And since the bloom filter is pure runtime and zero persistence, I figured we could just rand it at startup.

Reply:
As for the second question, I need to look into it more deeply. I don't recall it fully, on phone now

Reply:
Yep, it's totally fine to keep using the different offset for account data and storage data, just curious about the rationale behind.

Reply:
> I can understand we must choose different offsets for account and destructs,
> but what if we remove the destructs, is it still necessary to select a different offset
> for account data and storage data?

No, I'd say there's no need for that